lib/msal-common-poc/src/app/MsalConfiguration.ts

```diff
@@ -0,0 +1,141 @@
+ * - storage: this is where you configure storage implementation.
+ * - network: this is where you can configure network implementation.
+ */
+export type MsalConfiguration = {
+    auth: AuthOptions,
+    storageInterface: ICacheStorage,
+    networkInterface: INetworkModule,
+    cryptoInterface: ICrypto
+};
+
+const DEFAULT_AUTH_OPTIONS: AuthOptions = {
+    clientId: "",
+    clientSecret: "",
+    authority: null,
+    validateAuthority: true,
+    redirectUri: (): string => {
+        throw ClientAuthError.createRedirectUriEmptyError();
+    },
+    postLogoutRedirectUri: () => {
+        throw ClientAuthError.createPostLogoutRedirectUriEmptyError();
+    },
+    navigateToLoginRequestUrl: true
+};
+
+const DEFAULT_STORAGE_OPTIONS: ICacheStorage = {
+    clear: () => {
+        console.log("clear() has not been implemented for the cacheStorage interface.");
+    },
+    containsKey: (key: string): boolean => {
+        console.log("containsKey() has not been implemented for the cacheStorage interface.");
+        return false;
+    },
+    getItem: (key: string): string => {
+        console.log("getItem() has not been implemented for the cacheStorage interface.");
+        return "";
+    },
+    getKeys: (): string[] => {
+        console.log("getKeys() has not been implemented for the cacheStorage interface.");
+        return null;
+    },
+    removeItem: (key: string) => {
+        console.log("removeItem() has not been implemented for the cacheStorage interface.");
+        return;
+    },
+    setItem: (key: string, value: string) => {
+        console.log("setItem() has not been implemented for the cacheStorage interface.");
+        return;
+    }
+};
+
+const DEFAULT_NETWORK_OPTIONS: INetworkModule = {
+    sendRequestAsync: (url: string, method: RequestInit, enableCaching?: boolean): Promise<any> => {
+        console.log("Network interface - sendRequestAsync() has not been implemented");
+        return null;
+    }
+};
+
+const DEFAULT_CRYPTO_IMPLEMENTATION: ICrypto = {
+    base64Decode: (input: string): string => {
+        console.log("Crypto interface - base64Decode() has not been implemented");
+        return "";
+    },
+    base64Encode: (input: string): string => {
+        console.log("Crypto interface - base64Encode() has not been implemented");
+        return "";
+    },
+    generatePKCECodes(): Promise<PKCECodes> {
+        console.log("Crypto interface - generatePKCECodes() has not been implemented");
+        return null;
+    }
+};
+
+/**
+ * Function that sets the default options when not explicitly configured from app developer
+ *
+ * @param TAuthOptions
+ * @param TStorageOptions
+ * @param TSystemOptions
+ * @param TFrameworkOptions
+ *
+ * @returns TConfiguration object
+ */
+export function buildConfiguration({ auth, storageInterface, networkInterface, cryptoInterface }: MsalConfiguration): MsalConfiguration {
+    const overlayedConfig: MsalConfiguration = {
```

Comment:
do we think configuration between server and browser will use the same common object?

Reply:
I think we will have different auth options for server and browser, but everything else will be the same. So I am separating that into two different config objects in PR #1109.